transports/tcp/src/lib.rs

```diff
@@ -180,6 +184,8 @@ where
             nodelay: None,
             backlog: 1024,
             port_reuse: PortReuse::Disabled,
+            send_buffer_size: None,
```

Comment:
The doc comment of `fn new` would need updating as well. Wording should be similar to `IP_TTL`.